### PR TITLE
Roll up dependencies through TerminalApp so the package is right

### PR DESCRIPTION
**Pull request by DHowett-MSFT**
This pull request removes some of our manual cross-project reference hacks. CascadiaPackage no longer needs to import `Microsoft.UI.Xaml` directly to roll up its content, and it can get a bunch of stuff from TerminalApp (DLL only!) for free.

I've unpacked the package and verified that `resources.pri` contains TerminalApp's resources _and_ that the AppX manifest contains all of the activatable classes.

### CODE REVIEW

**Comment by DHowett-MSFT** [comment]
I have a crazy idea. I know how to stop this from regressing.

File: build/scripts/Test-WindowsTerminalPackage.ps1 [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    If ($LastExitCode -Ne 0) {
+        Throw "Failed to dump PRI"
+    }
+
+    $Manifest = [xml](Get-Content "$AppxPackageRootPath\AppxManifest.xml")
+    $PRIFile = [xml](Get-Content "$AppxPackageRootPath\resources.pri.xml")
+
+    ### Check the activatable class entries for a few DLLs we need.
+    $inProcServers = $Manifest.Package.Extensions.Extension.InProcessServer.Path
+    $RequiredInProcServers = ("TerminalApp.dll", "TerminalControl.dll", "TerminalConnection.dll")
```
Comment (zadjii-msft) [inline build/scripts/Test-WindowsTerminalPackage.ps1]:
```suggestion
    $RequiredInProcServers = ("TerminalApp.dll", "TerminalControl.dll", "TerminalConnection.dll", "TerminalSettings.dll")
```

**Comment by zadjii-msft** [comment]
I really like the simple validation script, good idea

File: build/scripts/Test-WindowsTerminalPackage.ps1 (hunk 1 above)
Comment (DHowett-MSFT) [reply build/scripts/Test-WindowsTerminalPackage.ps1]: intentional omission actually; we will eventually remove this dll & if Control is included we can use its existence as a proxy that "things are working okay"

**Comment by DHowett-MSFT** [comment]
By the way, this is what it looks like when the script fails!

```
% .\Test-WindowsTerminalPackage.ps1 CascadiaPackage_0.1.1211.0_x64.msix -verbose
VERBOSE: Expanding CascadiaPackage_0.1.1211.0_x64.msix
VERBOSE: Expanded to Temp\tmp95F4.tmp_Package
Output File at Temp\tmp95F4.tmp_Package\resources.pri.xml
Successfully Completed
VERBOSE: InProc Servers: TerminalSettings.dll TerminalConnection.dll TerminalControl.dll
 Microsoft.UI.Xaml.Markup.dll TerminalApp.dll Microsoft.UI.Xaml.dll
Failed to find App.xbf (TerminalApp project) in resources.pri
At openconsole-master\build\scripts\Test-WindowsTerminalPackage.ps1:75 char:9
+         Throw "Failed to find App.xbf (TerminalApp project) in resour ...
+ FullyQualifiedErrorId : Failed to find App.xbf (TerminalApp project) in resources.pri
```

**Pull request merged**